File: tests/cpp/src/TestServer.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <cameo/api/cameo.h>
#include <iostream>
#include <string>
#include <vector>
#include <sstream>

using namespace std;
using namespace cameo;

int main(int, char *[]) {

	cout << "Create server" << endl;

	unique_ptr<Server> server = Server::create("tcp://localhost:11000");
	server->setTimeout(1000);
	server->init();

	cout << "Testing connection" << endl;

	if (server->isAvailable()) {
		cout << "Server available" << endl;
	}

	cout << "Configs" << endl;

	vector<App::Config> configs = server->getApplicationConfigs();

	for (auto c : configs) {
		cout << c << endl;
	}

	try {
		unique_ptr<App> instance = server->start("simplecpp");
		state::Value state = instance->waitFor();

		cout << "Terminated simple with state " << toString(state) << endl;
	}
	catch (const StartException& e) {
		cout << "Cannot start application" << endl;
	}

	return 0;
}

File: tests/java/src/eu/ill/cameo/test/TestServer.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.test;

import java.util.List;

import eu.ill.cameo.api.base.App;
import eu.ill.cameo.api.base.Server;
import eu.ill.cameo.api.base.StartException;
import eu.ill.cameo.api.base.State;

public class TestServer {

	public static void main(String[] args) {

		System.out.println("Create server");
		
		Server server = Server.create("tcp://localhost:11000", 0);
		server.setTimeout(100);
		server.init();
		
		System.out.println("Testing connection");
		
		if (server.isAvailable()) {
			System.out.println("Server available");
		}
		
		System.out.println("Configs");
		
		List<App.Config> configs = server.getApplicationConfigs();
		
		for (App.Config c : configs) {
			System.out.println("  " + c.toString());
		}
		
		try {
			App instance = server.start("simplejava");
		
			int state = instance.waitFor();
		
			System.out.println("Terminated simple with state " + State.toString(state));
		}
		catch (StartException e) {
			System.out.println("App does not exist");
		}
		
		server.terminate();
	}

}